pumps/mongo.go

```diff
@@ -553,7 +552,10 @@ func (m *MongoPump) AccumulateSet(data []interface{}) [][]interface{} {
 		}
 
 		// Skip this record if it is a graph analytics record, they will be handled in a different pump
-		if thisItem.IsGraphRecord() {
+		if thisItem.IsGraphRecord() && !forGraph {
```

Comment:
You can simplify this to
`if thisItem.IsGraphRecord() != forGraph {
    continue
}`

Reply:
Or XOR ( a ^ !b) if you want to be fancy and explicit.

Reply:
Yeah, imo while it is simplified, I wanted to be as verbose as possible to aid understanding.

Reply:
Honestly, I read this, and all I could think was "why not just compare the two not to be equal?". That made it more confusing, at least for me.

If you really think it's better, I'd at least have the variables in the same order.

Reply:
@buraksezer could you please weigh in?

Reply:
I think `forGraph` is bad here. It makes it harder to understand. We can reorganize `AccumulateSet` function. We can add a new function called `AccumulateSetForGraph` and it can share code with `AccumulateSet`. They call a common function called `accumulateSetCommon` or something?

Reply:
@buraksezer so we can't split it cause it's the exact same functionality, only thing is the `forGraph` determins if only graph records will be accumulated or only non graph record. I think @Aenimus solution is better, with a better named variable

Reply:
Maybe we could also rename the variable also?

Reply:
Yup done that.